lib/contracts/blockchain.js

```diff
@@ -240,13 +240,55 @@ class Blockchain {
   }
 
   deployContractFromObject(deployContractObject, params, cb) {
+    const self = this;
+    let hash;
+    let calledBacked = false;
+
+    function callback() {
+      if (calledBacked) {
+        return;
+      }
+      if (interval) {
+        clearInterval(interval);
+      }
+      calledBacked = true;
+      cb(...arguments);
+    }
+
+    // This interval is there to compensate for the event that sometimes doesn't get triggered when using WebSocket
+    // FIXME The issue somehow only happens when the blockchain node is started in the same terminal
+    const interval = setInterval(() => {
+      if (!hash) {
+        return; // Wait until we receive the hash
+      }
+      self.web3.eth.getTransactionReceipt(hash, (err, receipt) => {
+        if (!err && !receipt) {
```

Comment:
it also needs too check for `receipt.contractAddress`, we only do the callback if the address is avaiable, which means the contract has been deployed